spec/facter/facter_spec.rb

```diff
@@ -56,13 +59,31 @@ def stub_no_fact
       expect(Facter.resolve('')).to be_an_instance_of(Hash)
     end
 
+    context 'when user query in arguments' do
+      it 'returns a stable order hash' do
+        stub_facts([hash_fact])
+        expect(Facter.resolve('mountpoints')).to eq('mountpoints' => { '/' => {}, '/boot' => {} })
+      end
+    end
+
     context 'when user query and options in arguments' do
       it 'returns one resolved fact' do
         stub_facts([os_fact])
 
         expect(Facter.resolve('os.name --show-legacy')).to eq('os.name' => 'ubuntu')
       end
     end
+
+    context 'when there is no user query' do
+      let(:hash_fact_no_user_query) do
+        Facter::ResolvedFact.new('mountpoints', { '/boot' => {}, '/' => {} }, :core, '')
+      end
+
+      it 'returns a stable order hash' do
+        stub_facts([hash_fact_no_user_query])
+        expect(Facter.resolve('')).to eq('mountpoints' => { '/' => {}, '/boot' => {} })
```

Comment:
Might be good to verify it sorts the entire tree, like:

```ruby
`mountpoints' => {
  '/' => {  'device' => 'dev1', 'filesystem' => 'ext4' }, 
  '/boot' => { 'device' => 'dev2', 'filesystem' => 'ext4' }
}
```

And then have the stub return `device` and `filesystem` in the reverse order.